devops/nightly.yml

```diff
@@ -13,182 +19,275 @@ schedules:
+  jobs:
+  - job:
+    displayName: Linux
+    pool:
+      vmImage: 'ubuntu-20.04'
+    timeoutInMinutes: 120
+    strategy:
+      matrix:
+        GCC Debug:
+          CC: gcc
+          CXX: g++
+          CXXFLAGS:
+          BuildType: Debug
+          Asan: Off
+        GCC Release:
+          CC: gcc
+          CXX: g++
+          CXXFLAGS:
+          Asan: Off
+          BuildType: Release
+        Clang Debug:
+          CC: clang
+          CXX: clang++
+          CXXFLAGS: -stdlib=libstdc++
+          BuildType: Debug
+          Asan: Off
+        Clang Release:
+          CC: clang
+          CXX: clang++
+          CXXFLAGS: -stdlib=libstdc++
+          Asan: Off
+          BuildType: Release
+        Clang Debug (ASAN):
+          CC: clang
+          CXX: clang++
+          CXXFLAGS: -stdlib=libstdc++
+          BuildType: Debug
+          Asan: On
+        Clang Release (ASAN):
+          CC: clang
+          CXX: clang++
+          CXXFLAGS: -stdlib=libstdc++
+          BuildType: Release
+          Asan: On
+    steps:
+    - checkout: self
+
+    - script: |
+        set -eo pipefail
+        git submodule init
+        git submodule update --depth 1 --recursive
+      displayName: 'Checkout submodules'
+
+    - template: ./steps/linux/dependencies.yml
+
+    - task: CMake@1
+      displayName: 'CMake'
+      inputs:
+        cmakeArgs: |
+          .. -GNinja -DCMAKE_BUILD_TYPE=$(BuildType) -DCMAKE_CXX_FLAGS=$(CXXFLAGS) -DENABLE_ASSERTS=ON -DUSE_ASAN=$(Asan) -DVERONA_CI_BUILD=On -DRT_TESTS=ON
+
+    - script: |
+        set -eo pipefail
+        ninja
+      workingDirectory: build
+      displayName: 'Compile'
+
+    - template: ./steps/linux/test.yml
+      parameters:
+        WorkingDirectory: build
+
+- stage: LinuxFull
+  displayName: Linux (Verona + LLVM)
+  dependsOn: []
+  jobs:
+  - job:
+    displayName: Linux
+    pool:
+      vmImage: 'ubuntu-20.04'
+    timeoutInMinutes: 120
+    strategy:
+      matrix:
+        GCC Release:
+          CC: gcc
+          CXX: g++
+          CXXFLAGS:
+          Asan: Off
+          BuildType: Release
+        Clang Debug (ASAN):
+          CC: clang
+          CXX: clang++
+          CXXFLAGS: -stdlib=libstdc++
+          BuildType: Debug
+          Asan: On
+
+    variables:
+      - template: ./vars/anybuild-vars.yml
+      - template: ./vars/llvm-vars.yml
+
+    steps:
+    - checkout: self
+
+    - script: |
+        set -eo pipefail
+        git submodule init
+        git submodule update --depth 1 --recursive
+      displayName: 'Checkout submodules'
+
+    - template: ./steps/linux/dependencies.yml
+
+    - bash: |
+        set -euo pipefail
+
+        # drop an AnyBuild.json config file and use it to
+        #   - ignore the CMAKE_BUILD_PARALLEL_LEVEL variable
+        #   - limit the number of concurrent cache lookups
+        echo '{"ActionCache": { "MaxParallelLookups": 10, "IgnoredEnvVars": [ "CMAKE_BUILD_PARALLEL_LEVEL" ] }}' | tee AnyBuild.json
```

Comment:
We could check in this `AnyBuild.json` file instead of dynamically generating it here, whatever you prefer.

Reply:
Would this be shared between different pipelines (ci, nightly)?

Reply:
Yes, I expect that for all different pipelines we'd use the same AnyBuild configuration.

For example, `MaxParallelLookups` really depends on the size of the VM where the build is running.  But for all the current pipelines I'm aware of, 10 should be a reasonable setting.

Reply:
So perhaps this could be checked in as a separate json file and all users include the same.